test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13034,6 +13045,48 @@ def __init__(self, username=None, attrs={}, jobname=None):
         self.set_variable_list(self.username)
         self.set_sleep_time(100)
 
+    def set_attributes(self, a={}):
+        """
+        set attributes and custom attributes on this object.
+        custom attributes are used when converting attributes to CLI.
+        In case of Cray platform if 'Resource_List.vntype' is set
+        already the remove it and add vntype value to select statement.
+
+        :param a: Attribute dictionary
+        :type a: Dictionary
+        """
+        if isinstance(a, list):
+            a = OrderedDict(a)
+
+        self.attributes = OrderedDict(self.dflt_attributes.items() +
+                                      self.attributes.items() + a.items())
+
+        if self.platform == 'cray' or self.platform == 'craysim':
+            s = 'Resource_List.select' in a.keys()
+            v = 'Resource_List.vntype' in self.custom_attrs.keys()
+            if s and v:
+                ra = []
+                vntype = self.custom_attrs['Resource_List.vntype']
+                del self.custom_attrs['Resource_List.vntype']
+                select = a['Resource_List.select']
+                c = select.startswith('vntype=') or ':vntype=' in select
+                m = '+vntype=' in select or c
+                r = select.split('+')
+                if not m:
+                    for i in r:
+                        i = i + ":vntype=" + vntype
+                        ra.append(i)
+                    string = ''
+                    for l in ra:
+                        if string is '':
```

Comment:
I guess you are doing this to avoid '+' in the begining. Can this be written in some other way? or just strip '+' from the beginning once you have the string ready.